view/src/components/farms.js

```diff
@@ -205,6 +227,13 @@ class Farms extends Component {
     super(props);
 
     this.state = {
+      // Search states
+      data: "",
+      filteredData: [],
+      nameQuery: "",
+      locationQuery: "",
+      tagsQuery: [],
+      allFarmTags: [],
       // Farm states
       farms: "",
```

Comment:
do you ever use the farm state? It appears that it is being replaced by the data state (if thats the case, consider deleting the farms state)